app/services/keys/query_service.rb

```diff
@@ -65,7 +65,7 @@ def query_local_devices!
   def query_remote_devices!
     return if @account.devices_url.blank?
 
-    json = fetch_resource(@account.devices_url)
+    json = fetch_resource(@account.devices_url, @account.id)
```

Comment:
The second argument can be omitted, in which case the resource will be fetched “anonymously”. And when passed, the argument must be an account, not an account ID.